gst/edge/edge_sink.c

```diff
@@ -0,0 +1,344 @@
+
+  gstbasesink_class->start = gst_edgesink_start;
+  gstbasesink_class->render = gst_edgesink_render;
+  gstbasesink_class->set_caps = gst_edgesink_set_caps;
+
+  GST_DEBUG_CATEGORY_INIT (GST_CAT_DEFAULT,
+      GST_EDGE_ELEM_NAME_SINK, 0, "Edge sink");
+}
+
+/**
+ * @brief initialize the new element
+ */
+static void
+gst_edgesink_init (GstEdgeSink * self)
+{
+  self->host = g_strdup (DEFAULT_HOST);
+  self->port = DEFAULT_PORT;
+}
+
+/**
+ * @brief set property
+ */
+static void
+gst_edgesink_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+
+  switch (prop_id) {
+    case PROP_HOST:
+      gst_edgesink_set_host (self, g_value_get_string (value));
+      break;
+    case PROP_PORT:
+      gst_edgesink_set_port (self, g_value_get_uint (value));
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief get property
+ */
+static void
+gst_edgesink_get_property (GObject * object, guint prop_id, GValue * value,
+    GParamSpec * pspec)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+
+  switch (prop_id) {
+    case PROP_HOST:
+      g_value_set_string (value, gst_edgesink_get_host (self));
+      break;
+    case PROP_PORT:
+      g_value_set_uint (value, gst_edgesink_get_port (self));
+      break;
+
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief finalize the object
+ */
+static void
+gst_edgesink_finalize (GObject * object)
+{
+  GstEdgeSink *self = GST_EDGESINK (object);
+  if (self->host) {
+    g_free (self->host);
+    self->host = NULL;
+  }
+
+  if (self->edge_h) {
+    nns_edge_release_handle (self->edge_h);
+    self->edge_h = NULL;
+  }
+
+  G_OBJECT_CLASS (parent_class)->finalize (object);
+}
+
+/**
+ * @brief start processing of edgesink
+ */
+static gboolean
+gst_edgesink_start (GstBaseSink * basesink)
+{
+  GstEdgeSink *self = GST_EDGESINK (basesink);
+
+  int ret;
+  char *port = NULL;
+  GstCaps *caps;
+
+  /** @todo support other connect_types */
+  nns_edge_connect_type_e connect_type = NNS_EDGE_CONNECT_TYPE_TCP;
+
+  ret =
+      nns_edge_create_handle ("TEMP_ID", connect_type,
+      NNS_EDGE_FLAG_SEND | NNS_EDGE_FLAG_SERVER, &self->edge_h);
+
+  if (NNS_EDGE_ERROR_NONE != ret) {
+    GST_ERROR ("Failed to get nnstreamer edge handle.");
+
+    if (self->edge_h) {
+      nns_edge_release_handle (self->edge_h);
+      self->edge_h = NULL;
+    }
+
+    return FALSE;
+  }
+
+  nns_edge_set_info (self->edge_h, "HOST", self->host);
+  port = g_strdup_printf ("%d", self->port);
+  nns_edge_set_info (self->edge_h, "PORT", port);
+  g_free (port);
+
+  caps = gst_pad_get_current_caps (GST_BASE_SINK_PAD (basesink));
+  if (!caps) {
+    caps = gst_pad_peer_query_caps (GST_BASE_SINK_PAD (basesink), NULL);
+  }
+
+  gst_caps_unref (caps);
```

Comment:
These lines are meaningless.
There are same unnecessary lines in the query server sink, so please delete it with another PR.

Reply:
Updated.
Releated PR(https://github.com/nnstreamer/nnstreamer/pull/3876)